### PR TITLE
Earth wall rework : Part 1

### PR DESCRIPTION
**Pull request by Aang23**
You already know what this PR is about. This first part implements :
- A way to place the wall in-place permanently,
- The wall now forms 2 (by default) blocks in front on the player, or 1 || 2 (by default, again), lower., even if the bender wasn't targeting a block.

PS : I hope my StatusControl texture isn't too bad...

Edit : I guess once merged I'll do a small import cleanup.

### CODE REVIEW

File: src/main/java/com/crowsofwar/avatar/common/bending/earth/AbilityWall.java [hunk 1]
```diff
@@ -96,7 +101,19 @@ public void execute(AbilityContext ctx) {
...
 				lookPos = lookPos.down(2);
 			}
-			if (STATS_CONFIG.bendableBlocks.contains(lookBlock) || STATS_CONFIG.plantBendableBlocks.contains(lookBlock)) {
+
+			// Allow bending even if the block is lower than the bender by 1-2 (by default)
+			// blocks
+			if (lookBlock == Blocks.AIR) {
+				for (int i = 0; i <= reach; i++) {
+					lookPos = lookPos.down();
+					if (world.getBlockState(lookPos).getBlock() != Blocks.AIR)
```
Comment (Mahtaran) [inline src/main/java/com/crowsofwar/avatar/common/bending/earth/AbilityWall.java]: Shouldn’t `lookBlock` be set to this new value?

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java [hunk 2]
```diff
@@ -158,6 +158,16 @@ public boolean isShield() {
 		return true;
 	}
 
+	/*
+	 * Changes how the wall should drop its blocks. false -> dropBlocks(), true ->
+	 * placeBlocks(). dropBlocks() is the default
+	 */
+	public void setDropTypePlace(Boolean input) {
```
Comment (Mahtaran) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: `Boolean` should be changed to `boolean`

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java [hunk 3]
```diff
@@ -158,6 +158,16 @@ public boolean isShield() {
...
+		dropTypePlace = input;
+	}
+
+	boolean dropTypePlace = false;
```
Comment (Mahtaran) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: Should probably be made private

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java (hunk 2 above)
Comment (Aang23) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: Ok, just a typo there.

File: src/main/java/com/crowsofwar/avatar/common/bending/earth/AbilityWall.java (hunk 1 above)
Comment (Aang23) [reply src/main/java/com/crowsofwar/avatar/common/bending/earth/AbilityWall.java]: I wasn't sure but I think you're right.

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java (hunk 3 above)
Comment (Aang23) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: No problem.

**Comment by Aang23** [comment]
No I wasn't... If that's an issue, I can use it.

**Comment by Mahtaran** [comment]
Please do use it if you don’t mind

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java (hunk 2 above)
Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: _That's what real IDE's are for_

**Comment by FavouriteDragon** [comment]
Looks good, but stuff like ```if (dropTypePlace)
					entity.placeBlocks();
				else
					entity.dropBlocks();
```
hurts me on the inside xD. Single if statements without curly braces are fine, but whole if else statements should have curly braces. It just makes them easier to read.
Otherwise, it looks good!

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java [hunk 4]
```diff
@@ -166,7 +176,11 @@ public void setDead() {
 			if (entity != null) {
 				// Avoid setDead() as that will call wall.setDead()
 				entity.isDead = true;
-				entity.dropBlocks();
+
+				if (dropTypePlace)
```
Comment (FavouriteDragon) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: Just make sure stuff like this has curly braces- it's easier to read and more satisfying.

Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: Or simplify it to
`dropTypePlace ? placeBlocks() : dropBlocks;`

Comment (Aang23) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityWall.java]: Ok. No problem.

**Pull request merged**